airbyte-integrations/connectors/source-pinterest/source_pinterest/source.py

```diff
@@ -0,0 +1,334 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+from abc import ABC
+from base64 import standard_b64encode
+from datetime import datetime
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import pendulum
+import requests
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream, HttpSubStream
+from airbyte_cdk.sources.streams.http.auth import Oauth2Authenticator
+
+from .utils import analytics_columns, to_datetime_str
+
+
+class PinterestStream(HttpStream, ABC):
+    url_base = "https://api.pinterest.com/v5/"
+    primary_key = "id"
+    data_fields = ["items"]
+    raise_on_http_errors = True
+
+    def __init__(self, config: Mapping[str, Any]):
+        super().__init__(authenticator=config["authenticator"])
+        self.config = config
+
+    @property
+    def start_date(self):
+        return self.config["start_date"]
+
+    @property
+    def window_in_days(self):
+        return 185  # Set window_in_days to maximum available date range
```

Comment:
I recommend making this no larger than 30 days to accommodate very high scale scenarios

Reply:
Updated